tests/utils/__init__.py

```diff
@@ -42,3 +42,15 @@ def dump_sv(stream, metrics, delimiter=",", header=True):
         writer = csv.writer(stream)
         for d in metrics:
             writer.writerow(list(d.values()))
+
+
+def clean_staging():
+    from dvc.fs.memory import MemoryFileSystem
+    from dvc.objects.stage import _STAGING_MEMFS_PATH
+
+    try:
+        MemoryFileSystem().fs.rm(
+            f"memory://{_STAGING_MEMFS_PATH}", recursive=True
+        )
+    except FileNotFoundError:
+        pass
```

Comment:
Would it make sense to ovverride fsspec's memfs so that we could have isolated instances?

Reply:
So the way this works now is that we do have isolated instances per-ODB. `objects.stage(odb, ...)` will return a refodb placed at `memory://_STAGING_MEMFS_PATH/<hash of destination ODB path>`. So in practical terms, staging ODB URLs are unique per repo local cache directory

The issue is that for testing purposes, even with unique staging ODBs, we will still have to manually clear what's in memfs for certain tests. In real-world CLI use, staging is cleared after every DVC command (when the process terminates), but in our tests we just repeatedly call `dvc.func()` which does not clear anything in memfs, since it's all done in the same process.

Reply:
Since we return the odb instance each time we `stage()` something, we could maybe just generate a new staging ODB path per `stage()` call (using uuids), but this would end up causing us to duplicate a lot of behavior internally, since we call `stage()` multiple times even in a single DVC command.

output.commit, output.save, checkout all end up staging the same path whenever we `dvc commit` something. So if each call used a unique ODB path, we would end up recalculating the entire staged dir each time instead of just re-using the staged ref objects from the initial call

Reply:
@pmrowla Feels like we'll probably end up with 1 staging odb per repo instance or something like that in the future, and so it makes sense that it looks like it does in this PR 👍 Let's keep it for now.